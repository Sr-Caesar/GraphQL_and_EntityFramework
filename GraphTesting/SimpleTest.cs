


using GraphQL_Exercice.GraphQLSchema.Mutations;
using GraphTesting;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.Configuration;

namespace Testing
{
    public class SimpleTest
    {
        [Fact]
        public async Task SchemaChangeTest()//CodeFirst or Anotation Based with schema as an artifact
        {
            var schema = await new ServiceCollection()
            .AddGraphQLServer()
            .AddQueryType<Query>()
            .BuildSchemaAsync();

            //var schema = await TestServices.Executor.GetSchemaAsync(default);

            schema.ToString().MatchSnapshot();
        }

        [Fact]//TODO https://www.youtube.com/watch?v=Nf7nX2H_iiM
        public async Task GetCompaniesTest()
        {
            var testBuilder = WebApplication.CreateBuilder(new string[] { });
            string myQuery = @"{
                  companies{
                    id
                    name
                    dateOfFundation
                  }
                }";
            var result = await new ServiceCollection()
            .AddDbContext<ExerciceData.Context.AppDbContext>(option =>
            {
                option.UseSqlServer(testBuilder.Configuration.GetConnectionString("DefaultConnection"));
            })
            .AddScoped<CompanyRepository>()
            .AddGraphQLServer()
            .AddQueryType<Query>()
            .ExecuteRequestAsync(myQuery);

            result.ToJson().MatchSnapshot();
        }
    }
}
