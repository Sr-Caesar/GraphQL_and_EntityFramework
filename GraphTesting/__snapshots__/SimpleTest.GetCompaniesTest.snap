{
  "data": {
    "companies": [
      {
        "id": 1,
        "name": "Company A",
        "dateOfFundation": "2020-01-01T00:00:00.000\u002B01:00"
      },
      {
        "id": 2,
        "name": "Company B",
        "dateOfFundation": "2015-05-15T00:00:00.000\u002B02:00"
      },
      {
        "id": 3,
        "name": "Company C",
        "dateOfFundation": "2018-10-10T00:00:00.000\u002B02:00"
      },
      {
        "id": 6,
        "name": "prova",
        "dateOfFundation": "2019-12-31T23:00:00.000\u002B01:00"
      },
      {
        "id": 7,
        "name": "NOME DECENTE",
        "dateOfFundation": "0001-01-01T00:00:00.000Z"
      }
    ]
  }
}
